### PR TITLE
SONARJAVA-1784 Deprecate unused complexity API

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java [hunk 1]
```diff
@@ -58,6 +58,10 @@
 
   List<Tree> getComplexityNodes(Tree tree);
 
+  @Deprecated
+  /**
+   * @deprecated use getComplexityNodes instead
+   */
   List<Tree> getMethodComplexityNodes(ClassTree enclosingClass, MethodTree methodTree);
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

File: java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java [hunk 2]
```diff
@@ -58,6 +58,10 @@
 
   List<Tree> getComplexityNodes(Tree tree);
 
+  @Deprecated
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java]: Please move the annotation after the javadoc.

File: java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java [hunk 3]
```diff
@@ -58,6 +58,10 @@
 
   List<Tree> getComplexityNodes(Tree tree);
 
+  /**
+   * @deprecated use getComplexityNodes instead
+   */
+  @Deprecated
   List<Tree> getMethodComplexityNodes(ClassTree enclosingClass, MethodTree methodTree);
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by benzonico** [comment]
Merged in master.

**Pull request closed without merging**